zebra-network/src/peer_set/initialize/tests/vectors.rs

```diff
@@ -587,6 +593,367 @@ async fn crawler_peer_limit_default_connect_ok_stay_open() {
     );
 }
 
+/// Test the listener with an inbound peer limit of zero peers, and a handshaker that panics.
+#[tokio::test]
+async fn listener_peer_limit_zero_handshake_panic() {
+    zebra_test::init();
+
```

Comment:
Tip: You can speed up these tests without breaking them by pausing time. With the Tokio update that would be to just change the attribute `#[tokio::test(start_paused = true)]`, but before the update we have to manually call `tokio::time::pause();`:

```suggestion
    zebra_test::init();
    tokio::time::pause();
```

Reply:
Using `pause` caused a few tests to run for a very long time:
```
test peer_set::initialize::tests::vectors::crawler_peer_limit_one_connect_error has been running for over 60 seconds                                                                           
test peer_set::initialize::tests::vectors::crawler_peer_limit_one_connect_ok_then_drop has been running for over 60 seconds                                                                    
test peer_set::initialize::tests::vectors::peer_limit_one_mainnet has been running for over 60 seconds                                                                                         
test peer_set::initialize::tests::vectors::peer_limit_one_testnet has been running for over 60 seconds                                                                                         
test peer_set::initialize::tests::vectors::peer_limit_two_mainnet has been running for over 60 seconds                                                                                         
test peer_set::initialize::tests::vectors::peer_limit_two_testnet has been running for over 60 seconds   
```

These tests normally finish in under 15 seconds on my machine.

Maybe we need to add some `task::yield`s or do some other fixes to make this work.
But since it's risky I don't want to do it in this PR.

Reply:
Oops, I only added the `tokio::time::pause()` line to tests after line 608, then I tested with `cargo test listener_peer`, which I didn't realize missed a few tests. Sorry about that :(

Very weird for it to hang. I can reproduce it here, I'll take a deeper look if I have some time. I wouldn't expect `task::yield` to fix it because _in theory_ Tokio's IO primitives (like `TcpStream`) should yield automatically after a certain time to avoid starvation. But there's definitely something really strange happening :thinking:

Reply:
I'd like to make these changes in a separate PR, so we can focus on getting the beta release tagged.